projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-expanding.spec.ts

```diff
@@ -0,0 +1,258 @@
+import { Component, ViewChild } from '@angular/core';
+import { async, TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { IgxTreeGridModule, IgxTreeGridComponent, IgxTreeGridRowComponent } from './index';
+import { IgxTreeGridExpandingComponent, IgxTreeGridPrimaryForeignKeyComponent } from '../../test-utils/tree-grid-components.spec';
+import { TreeGridFunctions } from '../../test-utils/tree-grid-functions.spec';
+import { configureTestSuite } from '../../test-utils/configure-suite';
+
+describe('IgxTreeGrid - Expanding/Collapsing actions', () => {
+    configureTestSuite();
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                IgxTreeGridExpandingComponent
+            ],
+            imports: [
+                BrowserAnimationsModule,
+                IgxTreeGridModule]
+        })
+        .compileComponents();
+    }));
+
+    it('check row expanding and collapsing are changing rows count (UI)', () => {
+        const fix = TestBed.createComponent(IgxTreeGridExpandingComponent);
```

Comment:
Creating the IgxTreeGridExpandingComponent should be separated in a beforeEach to avoid repeating the 3 lines of code in every test.